cmake: Enable standard warnings, install rule.

olifre opened the pull request:
Also set project-name for cmake to know.

olifre committed:
cmake: Enable standard warnings, install rule.

Also set project-name for cmake to know.

olifre commented:
Rebased onto master => fixes merge conflicts ;-).

jobisoft commented:
something is wrong, cannot merge, it looks like this pull request is based on the version without your commit #1 , compare:

https://github.com/jobisoft/jAnalyzer/blob/master/src/CMakeLists.txt
https://github.com/olifre/jAnalyzer/blob/cmake-stuff/src/CMakeLists.txt

your version still contains "Info" on line 8, which has be changed by your commit already

https://github.com/jobisoft/jAnalyzer/commit/f261345e146b8fc73d8fb9c6e2e8c92de1338aa4

what do we do now?

jobisoft commented:
oh, you fixed it! Did I do something wrong? What happend?

olifre commented:
My fault - on the other hand, I wanted to see what happens ;-). 
I had two separate pull request both modifying CMakeLists, in the same lines. When you merged the first one, the second one conflicted with the master version. 
So I rebased my branch on the new master, fixed the conflict, and pushed again => fixed. 
Nice to see this works in a clean way ;-).

jobisoft commented:
Nice. So for the future, if he tells me he cannot merge something, I 
just inform the contributor and wait. Did you see, that I could not 
merge (and tried to merge it manually via console and realized that the 
commit was not based on the latest version)?

Am 04.03.2015 um 22:56 schrieb olifre:

> My fault - on the other hand, I wanted to see what happens ;-).
> I had two separate pull request both modifying CMakeLists, in the same 
> lines. When you merged the first one, the second one conflicted with 
> the master version.
> So I rebased my branch on the new master, fixed the conflict, and 
> pushed again => fixed.
> Nice to see this works in a clean way ;-).
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/jobisoft/jAnalyzer/pull/4#issuecomment-77258365.

olifre commented:
> So for the future, if he tells me he cannot merge something, I just inform the contributor and wait.

Yes - since I "only" saw your message, no automatic mail that merging with master can not work anymore (which I thought would be sent....). 

> Did you see, that I could not merge (and tried to merge it manually via console and realized that the commit was not based on the latest version)?

I rebased (instead of merging) which is "almost" the same (it does not do the merge-commit but re-does all commits on top of the latest commit from master - this is only fine for stuff in pull requests or feature branches, since it rewrites all commits). But I had a look also here because I more or less expected a conflict ;-).
